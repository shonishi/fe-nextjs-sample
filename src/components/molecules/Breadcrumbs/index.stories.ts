import type { Meta, StoryObj } from '@storybook/react';
import Breadcrumbs from '.';

const meta: Meta<typeof Breadcrumbs> = {
  component: Breadcrumbs,
};

export default meta;
type Story = StoryObj<typeof Breadcrumbs>;

export const MultipleElements: Story = {
  args: {
    breadcrumbs: [
      { id: 1, name: 'おもちゃ', href: '#' },
      { id: 2, name: '赤ちゃん・幼児', href: '#' },
      { id: 3, name: 'お風呂用', href: '#' },
      { id: 4, name: 'あひるちゃん 5羽セット', href: '#' },
    ],
  },
};

export const SingleElements: Story = {
  args: {
    breadcrumbs: [{ id: 1, name: 'おもちゃ', href: '#' }],
  },
};

export const EmptyElements: Story = {
  args: {
    breadcrumbs: [],
  },
};
