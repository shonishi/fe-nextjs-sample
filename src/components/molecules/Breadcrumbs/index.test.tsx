import Breadcrumbs from '.';
import renderer from 'react-test-renderer';

describe('Breadcrumbs', () => {
  it('breadcrumbsの要素が複数の場合、全て表示されること', async () => {
    const tree = renderer
      .create(
        <Breadcrumbs
          {...{
            breadcrumbs: [
              { id: 1, name: 'おもちゃ', href: '#' },
              { id: 2, name: '赤ちゃん・幼児', href: '#' },
              { id: 3, name: 'お風呂用', href: '#' },
              { id: 4, name: 'あひるちゃん 5羽セット', href: '#' },
            ],
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('breadcrumbsの要素が1つの場合、1つだけ表示されること', async () => {
    const tree = renderer
      .create(
        <Breadcrumbs
          {...{ breadcrumbs: [{ id: 1, name: 'おもちゃ', href: '#' }] }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('breadcrumbsが空の場合、何も表示されないこと', async () => {
    const tree = renderer
      .create(<Breadcrumbs {...{ breadcrumbs: [] }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
