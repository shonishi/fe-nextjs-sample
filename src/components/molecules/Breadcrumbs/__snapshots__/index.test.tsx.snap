// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs breadcrumbsが空の場合、何も表示されないこと 1`] = `null`;

exports[`Breadcrumbs breadcrumbsの要素が1つの場合、1つだけ表示されること 1`] = `
<nav
  aria-label="Breadcrumb"
>
  <ol
    className="mx-auto flex max-w-2xl items-center space-x-2 px-4 sm:px-6 lg:max-w-7xl lg:px-8"
    role="list"
  >
    <li
      className="text-sm"
    >
      <a
        aria-current="page"
        className="font-medium text-gray-500 hover:text-gray-600"
        href="#"
      >
        おもちゃ
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs breadcrumbsの要素が複数の場合、全て表示されること 1`] = `
<nav
  aria-label="Breadcrumb"
>
  <ol
    className="mx-auto flex max-w-2xl items-center space-x-2 px-4 sm:px-6 lg:max-w-7xl lg:px-8"
    role="list"
  >
    <li>
      <div
        className="flex items-center"
      >
        <a
          className="mr-2 text-sm font-medium text-gray-900"
          href="#"
        >
          おもちゃ
        </a>
        <svg
          aria-hidden="true"
          className="h-5 w-4 text-gray-300"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 20"
          width={16}
        >
          <path
            d="M5.697 4.34L8.98 16.532h1.327L7.025 4.341H5.697z"
          />
        </svg>
      </div>
    </li>
    <li>
      <div
        className="flex items-center"
      >
        <a
          className="mr-2 text-sm font-medium text-gray-900"
          href="#"
        >
          赤ちゃん・幼児
        </a>
        <svg
          aria-hidden="true"
          className="h-5 w-4 text-gray-300"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 20"
          width={16}
        >
          <path
            d="M5.697 4.34L8.98 16.532h1.327L7.025 4.341H5.697z"
          />
        </svg>
      </div>
    </li>
    <li>
      <div
        className="flex items-center"
      >
        <a
          className="mr-2 text-sm font-medium text-gray-900"
          href="#"
        >
          お風呂用
        </a>
        <svg
          aria-hidden="true"
          className="h-5 w-4 text-gray-300"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 20"
          width={16}
        >
          <path
            d="M5.697 4.34L8.98 16.532h1.327L7.025 4.341H5.697z"
          />
        </svg>
      </div>
    </li>
    <li
      className="text-sm"
    >
      <a
        aria-current="page"
        className="font-medium text-gray-500 hover:text-gray-600"
        href="#"
      >
        あひるちゃん 5羽セット
      </a>
    </li>
  </ol>
</nav>
`;
